controllers/managedfusionoffering_controller.go

```diff
@@ -85,13 +97,118 @@ func pluginReconcile(r *ManagedFusionOfferingReconciler) (ctrl.Result, error) {
 	if _, found := r.managedFusionOffering.Spec.Config["onboardingValidationKey"]; !found {
 		return ctrl.Result{}, fmt.Errorf("%s CR does not contain onboardingValidationKey entry", r.managedFusionOffering.Name)
 	}
+	if err := reconcileStorageCluster(r); err != nil {
```

Comment:
- the func signature seems a bit different, is this intentional?

Reply:
reconcileStorageCluster was a reconciler function in deployer but in agent we are going to have plugin system where each offering can plug into this controller and do whatever is required to install and configure their offering. So we are passing reconciler to these reconcile functions.